src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -58,9 +58,14 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
         private static SegmentedList<VSCompletionItem> SortCompletionitems(AsyncCompletionSessionInitialDataSnapshot data, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            var items = new SegmentedList<VSCompletionItem>(data.InitialItemList);
-            items.Sort(VSItemComparer.Instance);
+            var items = new SegmentedList<VSCompletionItem>(data.InitialItemList.Count);
```

Comment:
Not sure if this is potentially slower to be adding them one-at-a-time instead of letting it do the copy (which under the covers can do a CopyTo operation). You still have to loop but maybe the copy operation was still faster the old way?

Reply:
Don't think so. I believe the way `data.InitialItemList` is implemented today would do an iteration regardless, since it need to ensure the underlying pooled object hasn't been reclaimed (this is an improvement can be done in the future for sure)